SCAutolib/models/gui.py

```diff
@@ -0,0 +1,281 @@
+
+    def __init__(self, directory: str):
+        self.directory = directory
+        self.screenshot_num = 1
+
+    def screenshot(self):
+        """
+        Runs ffmpeg to take a screenshot.
+
+        The filename of screenshot is then returned.
+        """
+
+        filename = f'/tmp/SC-tests/{self.screenshot_num}.png'
+        run(['ffmpeg', '-hide_banner', '-y', '-f', 'kmsgrab', '-i', '-', '-vf',
+             'hwdownload,format=bgr0', '-frames', '1', '-update', '1',
+             filename])
+
+        self.screenshot_num += 1
+        return filename
+
+
+class Mouse():
+    def __init__(self):
+        run(['modprobe', 'uinput'], check=True)
+        # initialize the uinput device
+        self.device = uinput.Device([
+            uinput.REL_X,
+            uinput.REL_Y,
+            uinput.BTN_LEFT,
+            uinput.BTN_MIDDLE,
+            uinput.BTN_RIGHT,
+            uinput.REL_WHEEL,
+        ])
+
+        self.MAX_RES = 10000
+        self.FLICK_TIME = 0.1
+        self.STEP_TIME = 0.005
+
+    def move(self, x: int, y: int):
+        """Moves the mouse cursor to specified absolute coordinate."""
+
+        for uinput_axis, value in [(uinput.REL_X, x), (uinput.REL_Y, y)]:
+            # Go all the way up/left with the mouse
+            sleep(self.FLICK_TIME)
+            self.device.emit(uinput_axis, -self.MAX_RES)
+            sleep(self.FLICK_TIME)
+
+            # Go to the exact coordinate
+            for i in range(value):
+                self.device.emit(uinput_axis, 1)
+                sleep(self.STEP_TIME)
+
+    def click(self, button: str = 'left'):
+        """
+        Clicks the any button of the mouse.
+
+        Button value can be 'left', 'right' or 'middle'.
+        """
+
+        button_map = {
+            'left': uinput.BTN_LEFT,
+            'right': uinput.BTN_RIGHT,
+            'middle': uinput.BTN_MIDDLE,
+        }
+        uinput_button = button_map[button]
+
+        # press the button
+        self.device.emit(uinput_button, 1)
+        # wait a little
+        sleep(0.1)
+        # release the button
+        self.device.emit(uinput_button, 0)
+
+
+class KB():
+    """Wrapper class for keyboard library."""
+
+    def __init__(self):
+        def kb_decorator(fn):
+            def wrapper(*args, **kwargs):
+                print('keyboard...')
+                fn(*args, **kwargs)
+                sleep(1)
+            return wrapper
+
+        # Workarounds for keyboard library
+        # keyboard.write types nothing if the delay is not set
+        self.write = kb_decorator(functools.partial(keyboard.write, delay=0.1))
+        self.send = kb_decorator(keyboard.send)
+        # For some reason the first keypress is never sent
+        # So this effectively does nothing
+        keyboard.send('enter')
+
+
+class GUI():
+    def __init__(self):
+        # Create the directory for screenshots
+        # TODO parametrize?
+        self.screenshot_directory = '/tmp/SC-tests'
```

Comment:
As pointed in review of tests, files in /tmp/SC-tests would be overwritten multiple tests run on the same machine consecutively. We need to distinguish between tests so we may want to create specific directories based for example on timestamps.